phpunit/functional/Glpi/Migration/GenericobjectPluginMigrationTest.php

```diff
@@ -0,0 +1,804 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace tests\units\Glpi\Migration;
+
+use Contract_Item;
+use DbTestCase;
+use Domain_Item;
+use DropdownTranslation;
+use FieldUnicity;
+use Glpi\Asset\AssetDefinition;
+use Glpi\Asset\Capacity\AllowedInGlobalSearchCapacity;
+use Glpi\Asset\Capacity\HasContractsCapacity;
+use Glpi\Asset\Capacity\HasDevicesCapacity;
+use Glpi\Asset\Capacity\HasDocumentsCapacity;
+use Glpi\Asset\Capacity\HasHistoryCapacity;
+use Glpi\Asset\Capacity\HasInfocomCapacity;
+use Glpi\Asset\Capacity\HasNetworkPortCapacity;
+use Glpi\Asset\Capacity\HasNotepadCapacity;
+use Glpi\Asset\Capacity\IsProjectAssetCapacity;
+use Glpi\Asset\Capacity\IsReservableCapacity;
+use Glpi\Dropdown\DropdownDefinition;
+use Glpi\Migration\GenericobjectPluginMigration;
+use Glpi\Migration\PluginMigrationResult;
+use Group_Item;
+use Infocom;
+use Profile;
+
+class GenericobjectPluginMigrationTest extends DbTestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        parent::setUpBeforeClass();
+
+        // Load it once for the whole class.
+        // This will create new tables, therefore cannot be executed inside the test DB transaction.
+        $queries = $DB->getQueriesFromFile(sprintf('%s/tests/fixtures/genericobject-migration/genericobject-db.sql', GLPI_ROOT));
+        foreach ($queries as $query) {
+            $DB->doQuery($query);
+        }
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        $tables = $DB->listTables('glpi\_plugin\_genericobject\_%');
+        foreach ($tables as $table) {
+            $DB->dropTable($table['TABLE_NAME']);
+        }
+
+        parent::tearDownAfterClass();
+    }
+
+    public function setUp(): void
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        parent::setUp();
+
+        // Load it inside the test DB transaction to not have to clean it manually
+        $queries = $DB->getQueriesFromFile(sprintf('%s/tests/fixtures/genericobject-migration/glpi-data.sql', GLPI_ROOT));
+        foreach ($queries as $query) {
+            $DB->doQuery($query);
+        }
+    }
+
+    public function testProcessMigration(): void
```

Comment:
This test is huge, maybe it could be split into multiple methods that each validate one "feature" ?

Reply:
The problem is that every step of the migration depends on each other. For instance, custom fields definition depends on assets/dropdows definitions (for foreign key fields), and assets/dropdown entries cannot be imported if the corresponding assets/dropdows definitions have not been imported first.

I do not know how to do it, unless by doing the same migration multiple time and just test a subset of data on each test method.